src/sentry/search/snuba/executors.py

```diff
@@ -93,112 +89,167 @@ def query(
+
+        selected_columns = []
+        if get_sample:
+            query_hash = md5(repr(conditions)).hexdigest()[:8]
+            selected_columns.append(
+                ("cityHash64", ("'{}'".format(query_hash), self.TABLE_ALIAS + "issue"), "sample")
+            )
+            sort_field = "sample"
+            orderby = [sort_field]
+            referrer = "search_sample"
         else:
-            conditions.append(converted_filter)
-
-    extra_aggregations = dependency_aggregations.get(sort_field, [])
-    required_aggregations = set([sort_field, "total"] + extra_aggregations)
-    for h in having:
-        alias = h[0]
-        required_aggregations.add(alias)
-
-    aggregations = []
-    for alias in required_aggregations:
-        aggregations.append(aggregation_defs[alias] + [alias])
-
-    if cursor is not None:
-        having.append((sort_field, ">=" if cursor.is_prev else "<=", cursor.value))
-
-    selected_columns = []
-    if get_sample:
-        query_hash = md5(repr(conditions)).hexdigest()[:8]
-        selected_columns.append(("cityHash64", ("'{}'".format(query_hash), "issue"), "sample"))
-        sort_field = "sample"
-        orderby = [sort_field]
-        referrer = "search_sample"
-    else:
-        # Get the top matching groups by score, i.e. the actual search results
-        # in the order that we want them.
-        orderby = ["-{}".format(sort_field), "issue"]  # ensure stable sort within the same score
-        referrer = "search"
-
-    snuba_results = snuba.dataset_query(
-        dataset=Dataset.Events,
-        start=start,
-        end=end,
-        selected_columns=selected_columns,
-        groupby=["issue"],
-        conditions=conditions,
-        having=having,
-        filter_keys=filters,
-        aggregations=aggregations,
-        orderby=orderby,
-        referrer=referrer,
-        limit=limit,
-        offset=offset,
-        totals=True,  # Needs to have totals_mode=after_having_exclusive so we get groups matching HAVING only
-        turbo=get_sample,  # Turn off FINAL when in sampling mode
-        sample=1,  # Don't use clickhouse sampling, even when in turbo mode.
-    )
-    rows = snuba_results["data"]
-    total = snuba_results["totals"]["total"]
-
-    if not get_sample:
-        metrics.timing("snuba.search.num_result_groups", len(rows))
-
-    return [(row["issue"], row[sort_field]) for row in rows], total
+            # Get the top matching groups by score, i.e. the actual search results
+            # in the order that we want them.
+            orderby = [
+                "-{}".format(sort_field),
+                self.TABLE_ALIAS + "issue",
+            ]  # ensure stable sort within the same score
+            referrer = "search"
+
+        snuba_results = snuba.dataset_query(
+            dataset=self.dataset,
+            start=start,
+            end=end,
+            selected_columns=selected_columns,
+            groupby=[self.TABLE_ALIAS + "issue"],
```

Comment:
I guess this should be changed to group_id as well. See lyn's PR mentioned above.